tests/test_reserveSourcesTask.py

```diff
@@ -0,0 +1,154 @@
+        self.num = 100  # Number of sources
+
+    def construct(self, name, fraction):
+        """Construct the test environment
+
+        This isn't called 'setUp' because we want to vary the `fraction`.
+
+        Parameters
+        ----------
+        name : `str`
+            Name of column for flagging reserved sources (without "_reserved").
+        fraction : `float`
+            Fraction of sources to reserve.
+
+        Return struct elements
+        ----------------------
+        catalog : `lsst.afw.table.SourceCatalog`
+            Catalog of sources.
+        task : `lsst.meas.algorithms.ReserveSourcesTask`
+            Task to do the reservations.
+        key : `lsst.afw.table.Key`
+            Key to the flag column.
+        """
+        schema = lsst.afw.table.SourceTable.makeMinimalSchema()
+        config = lsst.meas.algorithms.ReserveSourcesConfig()
+        config.fraction = fraction
+        task = lsst.meas.algorithms.ReserveSourcesTask(columnName=name, schema=schema,
+                                                       doc="Documentation is good", config=config)
+        key = schema[name + "_reserved"].asKey()
+        catalog = lsst.afw.table.SourceCatalog(schema)
+        catalog.reserve(self.num)
+        for _ in range(self.num):
+            catalog.addNew()
+        return Struct(catalog=catalog, task=task, key=key)
+
+    def check(self, sources, task, key, fraction):
+        """Check that source reservation is working
+
+        We test that the source reservation works, that it depends
+        on the RNG seed and that thigns behave as we expect when there's
+        a prior selection.
+
+        Parameters
+        ----------
+        catalog : `lsst.afw.table.Catalog` or `list` of `lsst.afw.table.Record`
+            List of sources.
+        task : `lsst.meas.algorithms.ReserveSourcesTask`
+            Task to do the reservations.
+        key : `lsst.afw.table.Key`
+            Key to the flag column.
+        fraction : `float`
+            Fraction of sources to reserve.
+        """
+        numExpect = fraction*len(sources)
+
+        # No prior
+        results1 = task.run(sources, expId=1)
+        flagged1 = np.array([ss.get(key) for ss in sources])
+        self.assertEqual(flagged1.sum(), numExpect)
+        np.testing.assert_array_equal(results1.reserved, flagged1)
+        np.testing.assert_array_equal(results1.use, np.logical_not(flagged1))
+
+        # Second run with different seed
+        for ss in sources:
+            ss.set(key, False)
+        results2 = task.run(sources, expId=2)
+        flagged2 = np.array([ss.get(key) for ss in sources])
+        self.assertEqual(flagged1.sum(), numExpect)
+        np.testing.assert_array_equal(results2.reserved, flagged2)
+        np.testing.assert_array_equal(results2.use, np.logical_not(flagged2))
+        if numExpect > 0:
+            self.assertFalse(np.all(flagged1 == flagged2))  # Pretty unlikely they match given different seeds
```

Comment:
You can put your comments in the assert failure message (that includes `fraction`). (also below)

Though, in this particular case, I'm a bit uncomfortable about "pretty unlikely" in a unittest; that seems to be screaming out for a failure on a new compiler/OS.

Reply:
I don't know how else to test that different seeds produce different selections. I don't think it's as big a problem as you think, as the seeds are fixed --- we're not allowing it to run *completely* randomly. However, there is a small, non-zero probability that two seeds produce the same selection. For the worst case, (selecting 10 out of 100), I believe that probability is one in 17,310,309,456,440, i.e., "pretty unlikely".